src/InputManager.cpp

```diff
@@ -32,11 +33,24 @@ void InputManager::MouseInput()
 	cursorY = ypos;
 	
 }
+
+#include <iostream>
 void InputManager::KeyInput()
 {
-	
+
 	if(glfwGetKey(window, GLFW_KEY_ESCAPE) == GLFW_PRESS)
 	{
 		glfwSetWindowShouldClose(window, GL_TRUE);
 	}
+	for(int i = 0; i < scene->selected.size() && i < 9; i++)
+	{
+		static int oldState = GLFW_RELEASE;
+		int newState = glfwGetKey(window, GLFW_KEY_0+i+1);
+		if(newState == GLFW_RELEASE && oldState == GLFW_PRESS)
+		{
+			scene->selected[i] = scene->selected[i] ? 0 : 1;
```

Comment:
Easier to just make selected a property of Geometries. That way we don't have to maintain two vectors.